src/Persistence/Sql/Query.php

```diff
@@ -0,0 +1,339 @@
+    {
+        $this->dsql = $this->dsql->dsql()->mode('select')->option('exists')->field($this->dsql);
+
+        return $this;
+    }
+
+    public function getIterator(): iterable
+    {
+        try {
+            return $this->model->toQuery('select')->execute();
+        } catch (\PDOException $e) {
+            throw (new Exception('Unable to execute iteration query', 0, $e))
+                ->addMoreInfo('query', $this->getDebugQuery())
+                ->addMoreInfo('message', $e->getMessage())
+                ->addMoreInfo('model', $this->model)
+                ->addMoreInfo('scope', $this->model->scope()->toWords());
+        }
+    }
+
+    public function count($alias = null)
+    {
+        $this->initWhere($this->model->scope());
+
+        $this->dsql->reset('field')->field('count(*)', $alias);
+
+        return $this;
+    }
+
+    public function aggregate($fx, $field, $alias = null, $coalesce = false)
+    {
+        $field = is_string($field) ? $this->model->getField($field) : $field;
+
+        $this->initWhere();
+
+        if ($coalesce) {
+            $expr = "coalesce({$fx}([]), 0)";
+        } else {
+            $expr = "{$fx}([])";
+        }
+
+        if (!$alias && $field instanceof FieldSqlExpression) {
+            $alias = $fx . '_' . $field->short_name;
+        }
+
+        $this->dsql->reset('field')->field($this->dsql->expr($expr, [$field]), $alias);
+
+        return $this;
+    }
+
+    public function get()
+    {
+        return $this->dsql->get();
+    }
+
+    public function getRow()
+    {
+        return $this->dsql->getRow();
+    }
+
+    public function getOne()
+    {
+        return $this->dsql->getOne();
+    }
+
+    public function order($order, $desc = null)
+    {
+        return $this->dsql->order($order, $desc);
+    }
+
+    public function limit($limit, $offset = 0)
+    {
+        return $this->dsql->limit($limit, $offset);
+    }
+
+    public function initOrder($fields = [])
+    {
+        foreach ($fields as $order) {
+            if ($order[0] instanceof Expression) {
+                $this->order($order[0], $order[1]);
+            } elseif (is_string($order[0])) {
+                $this->order($this->model->getField($order[0]), $order[1]);
+            } else {
+                throw (new Exception('Unsupported order parameter'))
+                    ->addMoreInfo('model', $this->model)
+                    ->addMoreInfo('field', $order[0]);
+            }
+        }
+
+        return $this->dsql;
+    }
+
+    public function initLimit($limit)
+    {
+        if ($limit) {
+            $offset = $limit[1] ?? 0;
+            $limit = $limit[0] ?? null;
+
+            if ($limit || $offset) {
+                if ($limit === null) {
+                    $limit = PHP_INT_MAX;
+                }
+
+                $this->limit($limit, $offset ?? 0);
+            }
+        }
+
+        return $this->dsql;
+    }
+
+//     public function __call($method, $args)
```

Comment:
you was using this probably in debug, yes, we should probably not left it here - any specific impl. should be queried thru ->dsql->method()

Reply:
This converts Model::order array to dsql